src/main/java/cubicchunks/lighting/LightingManager.java

```diff
@@ -46,7 +46,8 @@ public LightingManager(ICubicWorld world) {
 
 		this.skylightCubeDiffuseProcessor = new SkyLightCubeDiffuseProcessor(world, "Sky Light Diffuse", 5);
 
-		this.firstLightProcessor = new FirstLightProcessor("First Light", world.getCubeCache(), 1);
+		// TODO: Setting GeneratorStage to null. Analyze effects.
+		this.firstLightProcessor = new FirstLightProcessor(null, "First Light", world.getCubeCache(), 1);
```

Comment:
generator stage is used only in canUpdateCube, which calls ICubeWorld.blocksExist. If blocksExist would take Predicate<Cube> you could do something like this:
```java
return cube.getWorld().blocksExist(cubeCenter, totalRadius, false, (cube)->!cube.getGenerationStage().isInitialStage());
```